pkg/reconciler/instances/rma/integration_action.go

```diff
@@ -322,3 +334,18 @@ func findLatestRevision(releases []*release.Release) *release.Release {
 
 	return release
 }
+
+func generateVmalertGroup(context *service.ActionContext, id, num string) int {
+	div := DefaultVMAlertGroupsNum
+
+	if num != "" && id != "" {
+		var err error
+		div, err = strconv.Atoi(num)
+		if err != nil {
+			context.Logger.Debugf("got error %s when converting string to int for configuration: %s, use its default value: %d", err, RmiVmalertGroupsNum, DefaultVMAlertGroupsNum)
+			div = DefaultVMAlertGroupsNum
+		}
+	}
+
+	return int(id[0]) % div
```

Comment:
Unfortunately this algorithm is not producing uniform (equal) distribution over the possible output values. Example with 4 groups:

```
Modulo 4

0 -> 0
1 -> 1
2 -> 2
3 -> 3
4 -> 0
5 -> 1
6 -> 2
7 -> 3
8 -> 0
9 -> 1
A (code 65 as int) -> 1
B -> 2
C -> 3
D -> 0
E -> 1
F -> 2

0: 4
1: 5
2: 4
3: 3
```

I think we should really use some pseudo randomizer to achieve uniform distribution regardless of group size.

Reply:
Hi, Benjamin. this is to get the remainder. n%m, its remainder is always between [0,m). and now n is a fixed first letter of instance id (its ascii value), so the reminder is always the same for the same instance id.